src/GeoJsonPages/GeoJsonContentHandler.php

```diff
@@ -4,18 +4,49 @@
 
 namespace Maps\GeoJsonPages;
 
+use Content;
+use Maps\MapsFactory;
+use Maps\Presentation\OutputFacade;
+use MediaWiki\Content\Renderer\ContentParseParams;
+use ParserOutput;
+
 class GeoJsonContentHandler extends \JsonContentHandler {
 
 	public function __construct( $modelId = GeoJsonContent::CONTENT_MODEL_ID ) {
 		parent::__construct( $modelId );
 	}
 
 	protected function getContentClass(): string {
-		return GeoJsonContent::class;
+		return version_compare( MW_VERSION, '1.38', '<' ) ? GeoJsonLegacyContent::class : GeoJsonContent::class;
 	}
 
 	public function makeEmptyContent(): GeoJsonContent {
-		return new GeoJsonContent( GeoJsonContent::newEmptyContentString() );
+		$class = $this->getContentClass();
+		return new $class( $class::newEmptyContentString() );
```

Comment:
Why is this change needed?

Reply:
It's to avoid having to repeat the version check here, to make sure it returns an object of the right class (either GeoJsonContent or GeoJsonLegacyContent). This way, it won't have to be changed when eventually the legacy content class is removed.